appeals/api/src/server/endpoints/documents/documents.service.js

```diff
@@ -33,12 +52,143 @@ export const getDocumentsForAppeal = async (appealId, sectionName = null) => {
+ * @param {string} documentId
+ * @returns {Promise<Document>}
+ */
+export const getDocumentForAppeal = async (documentId) => {
+	const document = await getDocumentById(documentId);
+	return document;
+};
+
+/**
+ * @param {DocumentApiRequest} upload
+ * @param {number} appealId
+ * @returns {Promise<{documents: BlobInfo[]}>}}
+ */
+export const addDocumentsToAppeal = async (upload, appealId) => {
+	const appeal = await getAppeal(appealId);
+	if (!appeal || appeal.reference == null) {
+		throw new Error('Case not found or has no reference');
+	}
+	const { blobStorageContainer, documents } = upload;
+	const documentsToSendToDatabase = mapDocumentsForDatabase(
+		appealId,
+		blobStorageContainer,
+		documents
+	);
+	const documentsFromDatabase = await upsertDocumentMetadata(
+		appealId,
+		appeal.reference,
+		documentsToSendToDatabase
+	);
+
+	return {
+		documents: mapDocumentsForBlobStorage(documentsFromDatabase, appeal.reference)
+	};
+};
+
+/**
+ * @param {DocumentVersionApiRequest} upload
+ * @param {number} appealId
+ * @param {string} documentId
+ * @returns {Promise<{documents: BlobInfo[]}>}}
+ */
+export const addVersionToDocument = async (upload, appealId, documentId) => {
+	const appeal = await getAppeal(appealId);
+	if (!appeal || appeal.reference == null) {
+		throw new Error('Case not found or has no reference');
+	}
+
+	const { blobStorageContainer, document } = upload;
+	const documentsToSendToDatabase = mapDocumentsForDatabase(appealId, blobStorageContainer, [
+		document
+	]);
+
+	const documentVersion = documentsToSendToDatabase[0];
+	const masterDocument = await getDocumentById(documentId);
+	const newVersionId = masterDocument.latestVersionId + 1;
+	const documentName = mapDocumentNameForStorageUrl(masterDocument.name);
+	const newDocumentVersion = await upsertDocumentVersion({
+		documentGuid: documentId,
+		fileName: documentName,
+		originalFilename: documentVersion.name,
+		mime: documentVersion.documentType,
+		size: documentVersion.documentSize,
+		version: newVersionId,
+		blobStoragePath: mapBlobPath(masterDocument.guid, appeal.reference, documentName, newVersionId),
+		blobStorageContainer: documentVersion.blobStorageContainer
+	});
+
+	await updateDocument(documentId, {
+		latestVersionId: newDocumentVersion.version
+	});
+
+	return {
+		documents: mapDocumentsForBlobStorage(
+			[newDocumentVersion.Document],
+			appeal.reference,
+			newVersionId
+		)
+	};
+};
+
 /**
  * Returns the current appeal by reference
  * @param {number} appealId
- * @returns {Promise<import("../appeals.js").RepositoryGetByIdResultItem|void>}
+ * @returns {Promise<RepositoryResult|void>}
  */
 const getAppeal = async (appealId) => {
 	const appeal = await appealRepository.getById(appealId);
 	return appeal;
 };
+
+/**
+ *
+ * @param {number} caseId
+ * @param {string} reference
+ * @param {DocumentMetadata[]} documents
+ * @returns {Promise<Document[]>}
+ */
+const upsertDocumentMetadata = async (caseId, reference, documents) => {
```

Comment:
There are 3 db interactions within this method, would it make the code more reliable if they were placed in a db transaction so that if any fail then no db updates are done and the db isn't left with a partial update?